### ISSUE TITLE
Settings JSON "buildkit" key ignored, buildkit is used without regard to this value

### ISSUE THREAD

**Issue by rsethc** [docker/for-mac#6848]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 6F1D724A-081C-49C2-B4CC-81D3DADA2096/20230521161423

### Expected behavior
With the following configuration,
```
{
  "features": {
    "buildkit": false
  }
}
```
and/or by providing following environment variable,
```
DOCKER_BUILDKIT=0
```
the buildkit builder should not be used.

### Actual behavior
Until the past few weeks, disabling `buildkit` via the configuration file editor in Docker Desktop was sufficient to disable the buildkit builder.

~~Until yesterday, even though the configuration file `"buildkit": false` seemed to have no effect on disabling buildkit anymore, I was able to disable it by specifying the `DOCKER_BUILDKIT=0` environment variable.~~ *During creation of a basic sample to demonstrate the issue for this report, I found that setting the environment variable does work, and I was losing this variable with Docker builds using `sudo` in my project where the problem was occurring. `sudo -E` solved this. However, I'm leaving the original text in to illustrate why the `"buildkit": false` configuration feature no longer working as expected is still disruptive.*

Currently, even with both the `"buildkit": false` in configuration, and `DOCKER_BUILDKIT=0` environment variable set, Docker refuses to use the legacy builder, and uses buildkit instead, which breaks my builds. Aside from it breaking my builds, I vastly prefer the way the legacy builder writes its output to console as buildkit tends to hide output and also discards output colors.

### Information

  - macOS Version: Ventura 13.3.1 (a)
  - Intel chip or Apple chip: M1
  - Docker Desktop Version: v4.19.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`
[diagnose_check_stderr.txt](https://github.com/docker/for-mac/files/11525023/diagnose_check_stderr.txt)
[diagnose_check_stdout.txt](https://github.com/docker/for-mac/files/11525025/diagnose_check_stdout.txt)

### Steps to reproduce the behavior

This issue is not about fixing buildkit (I'd still prefer not to use it even if it built my images successfully, just due to the console output appearance alone). Even if the reason why buildkit isn't working with my setup is potentially something trivial to fix, I still would consider it a bug that the configuration `"buildkit": false` setting is being ignored when I want to use the legacy builder.

  1. Use `docker build .` in a directory containing the following `Dockerfile`, after you've set `"buildkit": false` in the configuration JSON through the Docker Desktop menu.
  
```
FROM glassblade/slice:ubuntu
RUN echo "Hello"
```

  2. Observe that even though you set `"buildkit": false` in the configuration and then pressed `Apply & Restart` so that the Docker engine should default to not using buildkit, buildkit is still used, whereas the legacy builder should be seen.

Further context: `"buildkit": false` no longer having the effect it should is also a problem on Windows. I only started writing out an actual GitHub issue because when I tried this on Mac (where my build scripts have `sudo` which is not a thing on Windows), I thought my workflow was not just inconvenienced by having to disable buildkit another way, but permanently broken.

Sample output from running the above example Dockerfile in `sudo` (so that `DOCKER_BUILDKIT=0` environment variable is not inherited, and the configuration `"buildkit": false` ought to be using the legacy builder, but as you can see it does not, and buildkit fails to build) and without `sudo` (which shows the legacy builder, due to my environment having `DOCKER_BUILDKIT=0`, which I configured due to the configuration `"buildkit": false` being ignored):
```
seth@Seths-MacBook-Air sample % sudo docker build .
[+] Building 0.0s (0/1)                                                         
[+] Building 0.1s (3/3) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 83B                                        0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => ERROR [internal] load metadata for docker.io/glassblade/slice:ubuntu   0.0s
------
 > [internal] load metadata for docker.io/glassblade/slice:ubuntu:
------
Dockerfile:1
--------------------
   1 | >>> FROM glassblade/slice:ubuntu
   2 |     RUN echo "Hello"
   3 |     
--------------------
ERROR: failed to solve: error getting credentials - err: exit status 1, out: ``
seth@Seths-MacBook-Air sample % docker build .     
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.

Sending build context to Docker daemon  2.048kB
Step 1/2 : FROM glassblade/slice:ubuntu
 ---> dfee1f3065f4
Step 2/2 : RUN echo "Hello"
 ---> Using cache
 ---> 54be85dbb941
Successfully built 54be85dbb941
```

---

TL;DR: `"buildkit": false` in configuration JSON as seen in the Docker Desktop "Docker Engine" settings area should disable buildkit, and as of the past few weeks this setting seems to no longer have any effect.
